src/AdvApi32.Desktop/AdvApi32.Helpers.cs

```diff
@@ -14,29 +16,158 @@ namespace PInvoke
+        /// This string has a maximum length of 256 characters. The name is case-preserved in the service control manager.
+        /// Display name comparisons are always case-insensitive.
+        /// </param>
+        /// <param name="lpDescription">
+        /// The description of the service. If this member is NULL, the description remains unchanged.
+        /// If this value is an empty string (""), the current description is deleted.
+        /// The service description must not exceed the size of a registry value of type REG_SZ.
+        /// </param>
+        /// <param name="lpServiceStartName">
+        /// The name of the account under which the service should run.
+        /// Use an account name in the form DomainName\UserName. The service process will be logged on as this user.
+        /// If the account belongs to the built-in domain, you can specify .\UserName.
+        /// </param>
+        /// <param name="lpPassword">
+        /// The password to the account name specified by the lpServiceStartName parameter.
+        /// Specify an empty string if the account has no password or if the service runs in the LocalService, NetworkService, or LocalSystem account.
+        /// If the account name specified by the <paramref name="lpServiceStartName"/> parameter is the name of a managed service account or virtual account name, the lpPassword parameter must be NULL.
+        /// </param>
+        /// <exception cref="Win32Exception">If the method fails, returning the calling thread's last-error code value.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="lpServiceName" /> or <paramref name="lpBinaryPathName"/> are NULL or empty string.</exception>
+        /// <exception cref="FileNotFoundException"><paramref name="lpBinaryPathName"/> cannot be found.</exception>
+        public static void CreateService(string lpBinaryPathName, string lpServiceName, string lpDisplayName, string lpDescription, string lpServiceStartName, string lpPassword)
         {
-            using (SafeServiceHandle hSCManager = OpenSCManager(null, null, ServiceManagerAccess.SC_MANAGER_CREATE_SERVICE))
+            if (lpBinaryPathName == null)
             {
-                if (hSCManager.IsInvalid)
+                throw new ArgumentNullException(nameof(lpBinaryPathName));
+            }
+
+            if (lpBinaryPathName.Trim() == string.Empty)
+            {
+                throw new ArgumentException("Cannot be empty", nameof(lpBinaryPathName));
+            }
+
+            if (!File.Exists(lpBinaryPathName))
+            {
+                throw new FileNotFoundException("Cannot find the file.", lpBinaryPathName);
+            }
+
+            if (lpServiceName == null)
+            {
+                throw new ArgumentNullException(nameof(lpServiceName));
+            }
+
+            if (lpServiceName.Trim() == string.Empty)
+            {
+                throw new ArgumentException(nameof(lpServiceName));
```

Comment:
The single argument form of `ArgumentException` is taking a description (Where for `ArgumentNullException` it's a parameter name) :

```csharp
throw new ArgumentException("Service name must not be empty", nameof(lpServiceName));
```